tests/queries/test_explain.py

```diff
@@ -0,0 +1,100 @@
+import unittest
+
+from django.db import NotSupportedError, connection, transaction
+from django.db.models import Count
+from django.test import TestCase, skipIfDBFeature, skipUnlessDBFeature
+from django.test.utils import CaptureQueriesContext
+
+from .models import Tag
+
+
+@skipUnlessDBFeature('supports_explaining_query_execution')
+class ExplainTests(TestCase):
+
+    def test_basic(self):
+        querysets = [
+            Tag.objects.filter(name='test'),
+            Tag.objects.filter(name='test').select_related('parent'),
+            Tag.objects.filter(name='test').prefetch_related('children'),
+            Tag.objects.filter(name='test').annotate(Count('children')),
+            Tag.objects.filter(name='test').values_list('name'),
+            Tag.objects.order_by().union(Tag.objects.order_by().filter(name='test')),
+            Tag.objects.all().select_for_update().filter(name='test'),
+        ]
+        supported_formats = connection.features.supported_explain_formats
+        all_formats = (None,) + tuple(supported_formats) + tuple(f.lower() for f in supported_formats)
+        for idx, queryset in enumerate(querysets):
+            for format in all_formats:
+                with self.subTest(format=format, queryset=idx):
+                    if connection.vendor == 'mysql':
+                        # This does a query and caches the result.
+                        connection.features.needs_explain_extended
+                    with self.assertNumQueries(1), CaptureQueriesContext(connection) as captured_queries:
+                        result = queryset.explain(format=format)
+                        self.assertTrue(captured_queries[0]['sql'].startswith(connection.ops.explain_prefix))
+                        self.assertIsInstance(result, str)
+                        self.assertTrue(result)
+
+    @unittest.skipIf(connection.vendor == 'postgresql', 'Postgresql gives unrecognized EXPLAIN option "test"')
```

Comment:
3rd party backends might not want this test either. Add a database feature like `validates_explain_options`?